pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go

```diff
@@ -0,0 +1,236 @@
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	certificates, errs := gatherClusterIngressCertificates(ctx, gatherKubeClient.CoreV1(), operatorClient)
+
+	return []record.Record{{
+		Name: Filename,
+		Item: record.JSONMarshaller{Object: certificates},
+	}}, errs
+}
+
+func gatherClusterIngressCertificates(
+	ctx context.Context,
+	coreClient corev1client.CoreV1Interface,
+	operatorClient operatorclient.Interface) ([]*CertificateInfo, []error) {
+	//
+	var ingressAllowedNS = [2]string{"openshift-ingress-operator", "openshift-ingress"}
+
+	var certificatesInfo = make(map[CertificateInfoKey]*CertificateInfo)
+	var errs []error
+
+	// Step 1: Collect router-ca and router-certs-default
+	rCAinfo, err := getRouterCACertInfo(ctx, coreClient)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if rCAinfo != nil {
+		certificatesInfo[infereKey(rCAinfo)] = rCAinfo
+	}
+
+	rCDinfo, err := getRouterCertsDefaultCertInfo(ctx, coreClient)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if rCDinfo != nil {
+		certificatesInfo[infereKey(rCDinfo)] = rCDinfo
+	}
+
+	// Step 2: List all Ingress Controllers
+	for _, namespace := range ingressAllowedNS {
+		controllers, err := operatorClient.OperatorV1().IngressControllers(namespace).List(ctx, metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			klog.V(2).Infof("Ingress Controllers not found in '%s' namespace", namespace)
+			continue
+		}
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		// Step 3: Filter Ingress Controllers with spec.defaultCertificate and get certificate info
+		for i := range controllers.Items {
+			controller := controllers.Items[i]
+
+			// Step 4: Check the certificate limits
+			if len(certificatesInfo) >= ingressCertificatesLimits {
+				errs = append(errs, fmt.Errorf("reached the limit of ingress certificates (%d), skipping additional certificates", ingressCertificatesLimits))
+				break
+			}
+
+			if controller.Spec.DefaultCertificate != nil {
+				secretName := controller.Spec.DefaultCertificate.Name
+				certInfo, certErr := getCertificateInfoFromSecret(ctx, coreClient, namespace, secretName)
+				if certErr != nil {
+					errs = append(errs, certErr)
+					continue
+				}
+
+				// Step 5: Add certificate info to the certificates list
+				c, exists := certificatesInfo[infereKey(certInfo)]
+				if exists {
+					c.Controllers = append(c.Controllers,
+						ControllerInfo{Name: controller.Name, Namespace: controller.Namespace},
+					)
+					//
+				} else {
+					certInfo.Controllers = append(certInfo.Controllers,
+						ControllerInfo{Name: controller.Name, Namespace: controller.Namespace},
+					)
+					certificatesInfo[infereKey(certInfo)] = certInfo
+				}
+			}
+		}
+	}
+
+	var ci []*CertificateInfo
+	if len(certificatesInfo) > 0 {
+		// Step 6: Generate the certificates record
+		ci = make([]*CertificateInfo, len(certificatesInfo))
+		i := 0
+		for _, v := range certificatesInfo {
+			ci[i] = v
+			i++
```

Comment:
probably better to call just `ci = append(ci, v)`

Reply:
Solved in 82311f791c8e70eac909fe064f3a3eef081ea172